pyscriptjs/src/components/pyrepl.ts

```diff
@@ -167,11 +139,11 @@ export function make_PyRepl(interpreter: InterpreterClient) {
 
             // execute the python code
             // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-            const pyResult = (await pyExec(interpreter, pySrc, outEl)).result;
+            const { result } = await pyExec(interpreter, pySrc, outEl);
```

Comment:
why is it better? Does it have any functional difference or it's just more idiomatic? (I'm fine with more idiomatic of course, just curious)

Reply:
let's say once you know destructuring exist, it becomes natural to use destructuring wherever it helps, and these are the benefits in a nutshell:

  * **less refactoring**: if today `await thingAsObject` interesting field is just one but tomorrow you want to grab other properties, you just add the extra property you are after, as opposite of needing to assign the whole *await* in one line, then access `awaited.previousField` and `awaited.newInterstingField` ... with destructuring it's just `const { previousField, newInterstingField } = await sameThing`
  * **consistent style**: if destructuring is used in new code, new code should use it when appropriate to keep the code base consistent. Dare I say we might want to move everything into *destructuring* at some point, and I'd be happy to do so because of the previous advantage in the long term/future maintenance

Are these two points good enough to justify my choice there?

Reply:
maybe worth it as off topic: most of ECMAScript changes after ES2015 is about handy syntax shortcuts there due community demanding these and ease of refactoring or just easier on the eyes ... some syntax might look obscure but object destructuring and array destructuring are just blessing developers these days.

Reply:
I agree, I like object destructuring and I'd like to have something like that in Python. Thank for the explanation, +1 for keeping it